crates/revm/src/optimism/mod.rs

```diff
@@ -0,0 +1,173 @@
+            return Err(reth_executor::BlockExecutionError::L1BlockInfoError {
+                message: "unexpected l1 block info tx calldata length found".to_string(),
+            })
+        }
+
+        let l1_base_fee = U256::try_from_be_slice(&l1_info_tx_data[64..96]).ok_or(
+            reth_executor::BlockExecutionError::L1BlockInfoError {
+                message: "could not convert l1 base fee".to_string(),
+            },
+        )?;
+        let l1_fee_overhead = U256::try_from_be_slice(&l1_info_tx_data[192..224]).ok_or(
+            reth_executor::BlockExecutionError::L1BlockInfoError {
+                message: "could not convert l1 fee overhead".to_string(),
+            },
+        )?;
+        let l1_fee_scalar = U256::try_from_be_slice(&l1_info_tx_data[224..256]).ok_or(
+            reth_executor::BlockExecutionError::L1BlockInfoError {
+                message: "could not convert l1 fee scalar".to_string(),
+            },
+        )?;
+
+        Ok(Self { l1_base_fee, l1_fee_overhead, l1_fee_scalar })
+    }
+}
+
+impl L1BlockInfo {
+    /// Calculate the data gas for posting the transaction on L1. Calldata costs 16 gas per non-zero
+    /// byte and 4 gas per zero byte.
+    ///
+    /// Prior to regolith, an extra 68 non-zero bytes were included in the rollup data costs to
+    /// account for the empty signature.
+    pub fn data_gas(&self, input: &Bytes, chain_spec: Arc<ChainSpec>, timestamp: u64) -> U256 {
+        let mut rollup_data_gas_cost = U256::from(input.iter().fold(0, |acc, byte| {
+            acc + if *byte == 0x00 { ZERO_BYTE_COST } else { NON_ZERO_BYTE_COST }
+        }));
+
+        // Prior to regolith, an extra 68 non zero bytes were included in the rollup data costs.
+        if !chain_spec.fork(Hardfork::Regolith).active_at_timestamp(timestamp) {
+            rollup_data_gas_cost += U256::from(NON_ZERO_BYTE_COST).mul(U256::from(68));
+        }
+
+        rollup_data_gas_cost
+    }
+
+    /// Calculate the gas cost of a transaction based on L1 block data posted on L2
+    pub fn calculate_tx_l1_cost(
+        &self,
+        chain_spec: Arc<ChainSpec>,
+        timestamp: u64,
+        input: &Bytes,
+        is_deposit: bool,
+    ) -> U256 {
+        let rollup_data_gas_cost = self.data_gas(input, chain_spec, timestamp);
+
+        if is_deposit || rollup_data_gas_cost == U256::ZERO {
+            return U256::ZERO
+        }
+
+        rollup_data_gas_cost
+            .saturating_add(self.l1_fee_overhead)
+            .saturating_mul(self.l1_base_fee)
+            .saturating_mul(self.l1_fee_scalar)
+            .checked_div(U256::from(1_000_000))
+            .unwrap_or_default()
+    }
+}
+
+/// Get the base fee recipient address
+pub fn base_fee_recipient() -> Address {
+    Address::from_str(BASE_FEE_RECIPIENT).unwrap()
+}
+
+/// Get the L1 cost recipient address
+pub fn l1_cost_recipient() -> Address {
+    Address::from_str(L1_FEE_RECIPIENT).unwrap()
+}
```

Comment:
these should be static `Lazy<Address>`

Reply:
Resolved by your pr, tyvm (ref: https://github.com/anton-rs/op-reth/pull/98)